backend/courses/models.py

```diff
@@ -1419,3 +1419,40 @@ def create_or_update_user_profile(sender, instance, created, **kwargs):
     if created and instance.email != "":
         instance.profile.email = instance.email
     instance.profile.save()
+
+
+
+class Friendship(models.Model):
+    """
+    Used to track friendships along with requests status 
+    """
+    
+    requestSender = models.ForeignKey(
+        UserProfile,
+        on_delete=models.CASCADE,
+        related_name="sender",
+        help_text="The person (user) who sent the request.",
+    )
+
+    requestReciever = models.ForeignKey(
+        UserProfile,
+        related_name="receiver",
+         on_delete=models.CASCADE,
+        help_text="The person (user) who recieved the request.",
+    )
+
+    status = models.IntegerField(
+        help_text=dedent(
+            """
+        Whether the request was sent or accepted (0 if sent, 
+        1 if recieved and accepted, deleted if rejected)
+        """
+        ),
+    )
+
+    accepted_at = models.DateTimeField(auto_now_add=True)
```

Comment:
I don't think we can just use `auto_now_add` here if we make the enum change suggested in the comment above, unfortunately. Might need to override the save method: https://ilovedjango.com/django/models-and-databases/override-save-method-django-models/